### ISSUE TITLE
Not immutable Genotypes from GenotypeBuilder regarding extended attributes 

### ISSUE THREAD

**Issue by magicDGS** [samtools/htsjdk#580]
### Subject of the issue
When I try to re-use a `GenotypeBuilder` with an already initialized extended attribute map, the previously generated `Genotype`s contain the new attribute too. That means that a `Genotype` could be modified if you have access to the `GenotypeBuilder` that make it.

This is happening because the `extendedAttributes` field is passed by reference to the `FastGenotype` constructor. Changing the `FastGenotype` constructor will solve the problem.

### Your environment
* htsjdk 2.1.1
* Java(TM) SE Runtime Environment (build 1.8.0_60-b27)

### Steps to reproduce
```java
public class GenotypeBuilderFailTest {
	@Test
	public void test() {
		final GenotypeBuilder gb = new GenotypeBuilder("test");
		final int[] first = new int[]{1, 2};
		final int[] second = new int[]{3, 4};
		final Genotype firstG = gb.attribute("first", first).make();
		final Genotype secondG = gb.attribute("second", second).make();
		// both genotypes have the first field
		Assert.assertEquals(first, firstG.getExtendedAttribute("first"));
		Assert.assertEquals(first, secondG.getExtendedAttribute("first"));
		// only the second genotype should have the second field, but next line fails
		Assert.assertEquals(null, firstG.getExtendedAttribute("second"));
		Assert.assertEquals(second, secondG.getExtendedAttribute("second"));
	}
}
```

### Expected behaviour
A `Genotype` created by a `GenotypeBuilder` can't be modify if the builder is reused.

### Actual behaviour
A `Genotype` created by a `GenotypeBuilder` will contain extended attributes from later calls of `builder.attribute(key, value)`, unless the builder is reset.

**Comment by lindenb** [comment]
related to https://github.com/samtools/htsjdk/issues/541

**Comment by magicDGS** [comment]
Closed by #581.
